native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java

```diff
@@ -662,6 +661,7 @@ public void registerTestBinary(Project project,
             // Later this will be replaced by a dedicated task not requiring execution of tests
             testList.from(testListDirectory).builtBy(testTask);
             testOptions.getClasspath().from(testList);
+            testOptions.getRuntimeArgs().add("-D" + JUNIT_PLATFORM_LISTENERS_UID_TRACKING_OUTPUT_DIR + "=" + testResultsDir.dir(testTask.getName() + "/testlist").get().getAsFile().getAbsolutePath());
```

Comment:
```suggestion
            testOptions.getRuntimeArgs().add(testResultsDir.dir(testTask.getName() + "/testlist").map(d -> "-D" + JUNIT_PLATFORM_LISTENERS_UID_TRACKING_OUTPUT_DIR + "=" + d.getAsFile().getAbsolutePath()));

```

Reply:
Why is it done here and not in `TrackingDirectorySystemPropertyProvider` anymore? Actually I don't see this class changed either, is this redundant?

Reply:
Let me check if I can move it into `TrackingDirectorySystemPropertyProvider`. 

> is this redundant?

I would say it is not redundant. Without this line, we are not able to read the value of `UniqueIdTrackingListener.OUTPUT_DIR_PROPERTY_NAME` at runtime ([here](https://github.com/graalvm/native-build-tools/pull/693/files#diff-7c5f329e3e5f35f935274f283d925007ff37ed30f5a2bb2d3e7f4bf84304aa1eR155))